File: app/src/main/java/com/liubz/androidtea/TeaApplication.java
package com.liubz.androidtea;

import android.app.Activity;
import android.app.Application;
import android.app.NotificationChannel;
import android.app.NotificationManager;
import android.content.Context;
import android.os.Build;
import android.os.Bundle;
import android.util.Log;

import androidx.annotation.NonNull;
import androidx.annotation.Nullable;

import com.liubz.androidtea.network.websocket.WebSocketConfig;
import com.liubz.androidtea.network.websocket.WebSocketHelper;
import com.liubz.androidtea.utils.ApplicationUtils;
import com.liubz.androidtea.utils.CrashHandler;

public class TeaApplication extends Application {
    private static final String TAG = "TeaApplication";


    @Override
    protected void attachBaseContext(Context base) {
        super.attachBaseContext(base);
        ApplicationUtils.initApplication(this);
    }

    @Override
    public void onCreate() {
        super.onCreate();

        initHelper();
        initConfig();
        CrashHandler.getInstance().init(this);

        registerActivityLifecycleCallbacks(new ActivityLifecycleCallbacks() {
            @Override
            public void onActivityCreated(@NonNull Activity activity, @Nullable Bundle savedInstanceState) {
                Log.i(TAG, activity.getClass().getSimpleName() + " onCreate");
            }

            @Override
            public void onActivityStarted(@NonNull Activity activity) {
                Log.i(TAG, activity.getClass().getSimpleName() + " onStart");
            }

            @Override
            public void onActivityResumed(@NonNull Activity activity) {
                Log.i(TAG, activity.getClass().getSimpleName() + " onResume");
            }

            @Override
            public void onActivityPaused(@NonNull Activity activity) {
                Log.i(TAG, activity.getClass().getSimpleName() + " onPause");
            }

            @Override
            public void onActivityStopped(@NonNull Activity activity) {
                Log.i(TAG, activity.getClass().getSimpleName() + " onStop");
            }

            @Override
            public void onActivitySaveInstanceState(@NonNull Activity activity, @NonNull Bundle outState) {

            }

            @Override
            public void onActivityDestroyed(@NonNull Activity activity) {
                Log.i(TAG, activity.getClass().getSimpleName() + " onDestroy");
            }
        });
    }

    private void initHelper() {
        WebSocketHelper.init(new WebSocketConfig());
    }

    private void initConfig() {
        WebSocketHelper.initConfig();
    }
}

File: app/src/main/java/com/liubz/androidtea/TeaApplication.kt
package com.liubz.androidtea

import android.app.Activity
import android.app.Application
import android.content.Context
import android.os.Bundle
import android.util.Log
import com.liubz.androidtea.di.koin.module.appModule
import com.liubz.androidtea.utils.ApplicationUtils
import com.liubz.androidtea.utils.CrashHandler
import dagger.hilt.android.HiltAndroidApp
import org.koin.android.ext.koin.androidContext
import org.koin.android.ext.koin.androidLogger
import org.koin.core.context.startKoin

/**
 * @Desc: 应用程序入口，标注 @HiltAndroidApp 触发 Hilt 的代码生成
 */
@HiltAndroidApp
class TeaApplication : Application() {

    companion object {
        private const val TAG = "TeaApplication"
    }

    override fun attachBaseContext(base: Context) {
        super.attachBaseContext(base)
        ApplicationUtils.initApplication(this)
    }

    override fun onCreate() {
        super.onCreate()

        CrashHandler.getInstance().init(this)

        registerActivityLifecycleCallbacks(object : ActivityLifecycleCallbacks {
            override fun onActivityCreated(activity: Activity, savedInstanceState: Bundle?) {
                Log.i(TAG, "${activity.javaClass.simpleName} onCreate")
            }

            override fun onActivityStarted(activity: Activity) {}
            override fun onActivityResumed(activity: Activity) {}
            override fun onActivityPaused(activity: Activity) {}
            override fun onActivityStopped(activity: Activity) {}
            override fun onActivitySaveInstanceState(activity: Activity, outState: Bundle) {}
            override fun onActivityDestroyed(activity: Activity) {}
        })

        // Initialize Koin
        startKoin {
            androidLogger()
            androidContext(this@TeaApplication)
            modules(appModule)
        }
    }
}
